lib/query.js

```diff
@@ -2609,12 +2634,43 @@ Query.prototype._countDocuments = async function _countDocuments() {
   applyGlobalMaxTimeMS(this.options, this.model);
   applyGlobalDiskUse(this.options, this.model);
 
-  const conds = this._conditions;
   const options = this._optionsForExec();
 
+  this._applyTranslateAliases(options);
+
+  const conds = this._conditions;
+
   return this._collection.collection.countDocuments(conds, options);
 };
 
+/*!
+ * If `translateAliases` option is set, call `Model.translateAliases()`
+ * on the following query properties: filter, projection, update, distinct.
+ */
+
+Query.prototype._applyTranslateAliases = function _applyTranslateAliases(options) {
+  let applyTranslateAliases = false;
+  if ('translateAliases' in this._mongooseOptions) {
+    applyTranslateAliases = this._mongooseOptions.translateAliases;
+  } else if (this.model?.schema?._userProvidedOptions?.translateAliases != null) {
+    applyTranslateAliases = this.model.schema._userProvidedOptions.translateAliases;
+  } else if (this.model?.base?.options?.translateAliases != null) {
+    applyTranslateAliases = this.model.base.options.translateAliases;
+  }
+  if (!applyTranslateAliases) {
+    return;
+  }
+
+  if (this.model && this.model.schema && Object.keys(this.model.schema.aliases).length > 0) {
+    this.model.translateAliases(this._conditions);
+    this.model.translateAliases(options.projection);
+    this.model.translateAliases(this._update);
```

Comment:
Would this handle update operators such as `$inc`, `$unset`?
If not, I think it's okay to ship this as is and maybe support this in a future version, but I think it's worth having a test that verifies that at least `translateAliases` doesn't break the updates that include those operators.

Reply:
Good point, we need to handle atomic update operators.

Reply:
Fixed in 198aec9